plugin_ide.ui/src/com/github/rustdt/ide/ui/editor/RustEditorActionContributor.java

```diff
@@ -28,6 +47,88 @@ protected RustOpenDefinitionOperation createOpenDefinitionOperation(ITextEditor
 	
 	@Override
 	protected void registerOtherEditorHandlers() {
+		setupOpenType();
+	}
+	
+	// TODO: Evaluate source structure in background thread.

```

Comment:
I will be working on that. But if I could need some hints which infrastructure to reuse.